litejpeg/core/new_dct6.py

```diff
@@ -0,0 +1,504 @@
+# DCT
+import math
+
+from litex.gen import *
+from litex.soc.interconnect.stream import *
+
+from litejpeg.core.common import *
+
+'''
+This module is for the purpose of the DCT transformation.
+As the input given to this module is in the form of serial input of the value of approximately
+64*12 bits. This will go into the DCTdatapath where DCT1d conversion should take place on the 
+matrix formed after making 64 values of 12 bits each out of the serial input.
+Then the transpose of the resltant matrix is obtained and dct1D is again applied on this matrix.
+Finally the matrix which is been obtained is been converter into the serial output of 64*12 bits.
+'''
+datapath_latency = 8
+
+@CEInserter()
+class DCTDatapath(Module):
+
+    def __init__(self, dw, dct_block):
+        
+        #Record the input and the output of the Datapath.
+        # input = Sink.
+        # output = Source. 
+        self.sink = sink = Record(dct_block_layout(dw,dct_block))
+        self.source = source = Record(dct_block_layout(dw,dct_block))
+
+         # # #
+
+        dct_matrix_1d = Array(Array(Signal(dw) for a in range(8)) for b in range(8))
+        #dct_matrix_2d = Array(Array(Signal(dw) for a in range(8)) for b in range(8))
+
+        #intialise the vectors for 8 blocks at a time that will undergo DCT processing
+        #at a time.
+        vector0 = Array(Signal(dw) for a in range(8))
+        vector1 = Array(Signal(dw) for a in range(8))
+        vector2 = Array(Signal(dw) for a in range(8))
+        vector3 = Array(Signal(dw) for a in range(8))
+        vector4 = Array(Signal(dw) for a in range(8))
+        vector5 = Array(Signal(dw) for a in range(8))
+        vector6 = Array(Signal(dw) for a in range(8))
+        vector7 = Array(Signal(dw) for a in range(8))
+
+        vector_out = Array(Signal(dw) for a in range(64))
```

Comment:
Following are the vectors with their functions:

1. **"vector"+str(i) for i in range(8)** : 8 blocks with 8 bits each.
Taking its values from the dct_dalayed  passed as an input to the **DCT1d** module.

2. **vector_out** :  64 blocks with 8 bits each
Take the output from the **DCT1d** module for each of the above input.

3. **vector2_str(i) for i in range(8)** :  Each has 8 blocks with 8 bits.
Transpose **vector_out** and take the values to pass as an input to the **DCT1d** module.

4. **vector2_out** : 64 blocks with 8 bits in each.
Contains the output from the **DCT1d** module from the above input.

5. **vector2_final** : 64 blocks with 8 bits each.
Transpose of the **vector2_out** matrix above.

Hope this will answer your question.